Demo/Demo/Views/CustomButton.swift

```diff
@@ -1,9 +1,15 @@
 import UIKit
 
 class CustomButton: UIButton {
+    var titleHolder: String?
+    // swiftlint:disable:next implicitly_unwrapped_optional
+    var activityIndicator: UIActivityIndicatorView!
+
     convenience init(title: String) {
         self.init(type: .system)
 
+        titleHolder = title
```

Comment:
We need some sort of holder so we can set the title to nothing when we are animating the spinner and then re-add the title when we are done with the animation. Having some sort of holder was the best way I found to achieve this and to avoid having the title and spinner overlay one another, but certainly open to other options if there is a better way to do this!